components/dada-parse/src/parser/code.rs

```diff
@@ -325,15 +332,27 @@ impl CodeParser<'_, '_> {
             tracing::debug!("identifier");
             Some(self.add(ExprData::Id(id), id_span))
         } else if let Some((word_span, word)) = self.eat(Number) {
```

Comment:
> We might want to do something mildly unconventional and parse -5 without any whitespace as a literal that is explicitly a signed integer.  

I don't know how to implement this. I can't do this in `parse_expr_0`, because `-` will always be consumed in either unary or binary expressions.